utils.rkt

```diff
@@ -16,7 +16,9 @@
   (discretization n
                   (lambda (x)
                     (parameterize ([bf-precision n])
-                      (bfcopy x)))
+                      (if (equal? (bigfloats-between 0.bf x) 1)
+                          (bf 0)
+                          (bfcopy x))))
```

Comment:
This hack solves the following bug:
When evaluating with bf-discretization and the final result is supposed to be 0.bf,
We get an interval [0.bf, +min.bf] in a high precision.
Our algorithm tries to round both endpoints to a lower precision,
But since +min.bf has a mantissa format of 1.000... - it rounds again to +min.bf in lower precision.
The result is: the rounding is faithful but not correct.

It is done in convert function because Rival by default returns lower endpoint if distance is 0, however
what if the interval is [-min.bf, 0.bf] - 0.bf should be returned instead of -min.bf.
This hack solves this issue